### ISSUE TITLE
[feature] Ensure that `tools.gnu:pkg_config` is discoverable in all conan v2 build helpers

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#12354]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

Currently `tools.gnu:pkg_config` is only used in `PkgConfig` class (and it's explicitly described like this in `conan config list`). It would be a good idea to extend capabilities of this conf so that when it's set, it ensures that this specific executable is found by standard usage of CMake, Autotools, Meson etc...

For example in CMake, standard usage is to call `find_package(PkgConfig)`. There is a documented CMake variable `PKG_CONFIG_EXECUTABLE` which can be overridden to force this module file to use executable pointed by this variable: https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#variables-affecting-behavior

**Comment by eli-schwartz** [comment]
> There is a documented CMake variable `PKG_CONFIG_EXECUTABLE` which can be overridden to force this module file to use executable pointed by this variable

Note that all 3 of CMake, Meson, and autotools will respect the environment variable `$PKG_CONFIG` -- cmake will override that with the cmake variable ~~prefixed by `CMAKE_`~~ suffixed by `_EXECUTABLE` because I can't read and also because why ever not specify that it is an executable, and Meson will override that if you have a machine file whose `[binaries]` section maps a value for pkg-config.

But just using the environment variable is simple and consistent.

**Comment by SpaceIm** [comment]
Yes that's probably the easiest solution. The problem is that in conan it would likely be something handled by `VirtualBuildEnv`, but for some reason I don't understand it's disabled by default in conan v1 client.

note: I'm not sure there is a `CMAKE_PKG_CONFIG` cmake variable, it's not documented

**Comment by memsharded** [comment]
> Yes that's probably the easiest solution. The problem is that in conan it would likely be something handled by VirtualBuildEnv, but for some reason I don't understand it's disabled by default in conan v1 client.

It is disabled by default in Conan 1.X, because it is a new thing that could break users. It is opt-in in two ways:
- Declaring ``generators = "VirtualBuildEnv"`` in the recipe
- Using the ``tools.env.virtualenv:auto_use=True`` conf for enabling it for all packages

**Comment by SpaceIm** [comment]
But in the meantime it "breaks" conan recipes relying on v2 helpers since env vars are not honored. `tools.env.virtualenv:auto_use=True` is really an internal detail of conan, users shouldn't have to know how these things work in conan.
